src/figuresClasses.ts

```diff
@@ -1,11 +1,77 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea: () => number;
+}
+
+function areaRound(area: number): number {
+  return Math.floor(area * 100) / 100;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: string = 'triangle',
+  ) {
+    const sides = [a, b, c].sort((side1, side2) => side1 - side2);
+    const isValidSides = sides[2] < sides[0] + sides[1];
+
+    if (a <= 0 || b <= 0 || c <= 0 || !isValidSides) {
+      throw new Error(
+        'Triangle sides must be positive and satisfy the triangle inequality',
+      );
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
 
-export class Triangle implements Figure {}
+    return areaRound(area);
+  }
+}
+
+export class Circle implements Figure {
+  constructor(
+    public color: string,
+    public radius: number,
+    public shape: string = 'circle',
+  ) {
+    if (radius <= 0) {
+      throw new Error('Circle radius must be a positive number');
+    }
```

Comment:
Recommend creating a separate function like this:
```
function checkSides(errorMessage: string, ...sides: number[]): void {
  if (sides.some(side => side <= 0)) {
    throw new Error(errorMessage);
  }
}
```
because this check is used three times